### PR TITLE
Feature hardware usart 

### PR DESCRIPTION
**Pull request by PatrickDekker98**
New pull request with new changes based on feedback of previous PR

### CODE REVIEW

File: libraries/hwlib/library/core/hwlib-i2c.hpp [hunk 1]
```diff
@@ -494,7 +494,7 @@ class i2c_channel {
 /// - <a href="https://en.wikipedia.org/wiki/I2C">
 ///    I2C Bus</A> (wikipedia)
 ///
-class i2c_bus : protected i2c_primitives {
+class i2c_bus : public i2c_primitives {
```
Comment (gerritvanos) [inline libraries/hwlib/library/core/hwlib-i2c.hpp]: why a i2c change in usart PR?

File: libraries/hwlib/library/targets/hwlib-arduino-due.hpp [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
 #define register
 #include "sam.h"
+//#include "usart_queue.hpp"
```
Comment (gerritvanos) [inline libraries/hwlib/library/targets/hwlib-arduino-due.hpp]: why is this commented?

File: libraries/hwlib/library/core/hwlib-i2c.hpp (hunk 1 above)
Comment (PatrickDekker98) [reply libraries/hwlib/library/core/hwlib-i2c.hpp]: While testing i found that this gave an error " i2c_bus is protected in this context" for i2c_bit_banged

File: libraries/hwlib/library/targets/hwlib-arduino-due.hpp (hunk 2 above)
Comment (PatrickDekker98) [reply libraries/hwlib/library/targets/hwlib-arduino-due.hpp]: slordigheid

File: libraries/hwlib/library/core/hwlib-i2c.hpp (hunk 1 above)
Comment (gerritvanos) [reply libraries/hwlib/library/core/hwlib-i2c.hpp]: allright

File: libraries/hwlib/library/targets/hwlib-arduino-due.hpp (hunk 2 above)
Comment (gerritvanos) [reply libraries/hwlib/library/targets/hwlib-arduino-due.hpp]: can you delete it then?

Comment (PatrickDekker98) [reply libraries/hwlib/library/targets/hwlib-arduino-due.hpp]: Done

**Pull request merged**